### PR TITLE
document & refactor `artifacts.get_artifacts()`

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR:
* refactor `artifacts.get_artifacts()`
* + adds docstring
* + adds type hints
* + adds small tests where no threshold found
* #20

### CODE REVIEW

File: sgeop/artifacts.py [hunk 1]
```diff
@@ -22,29 +22,74 @@
...
+
+    def _relate(neighs: tuple, cond: callable) -> bool:
+        """Helper for relating artifacts."""
+        return len(neighs) > 0 and cond(polys.loc[list(neighs), "is_artifact"])
+
     with warnings.catch_warnings():  # the second loop likey won't find threshold
         warnings.filterwarnings("ignore", message="No threshold found")
         fas = momepy.FaceArtifacts(roads)
-    polygons = fas.polygons.set_crs(roads.crs)
+    polys = fas.polygons.set_crs(roads.crs)
```
Comment (martinfleis) [inline sgeop/artifacts.py:84]: Out of curiosity, why?

**Review by martinfleis — APPROVED** [review]

**Comment by martinfleis** [comment]
Ah, too early to approve. Some failures caught by CI.

File: sgeop/artifacts.py (hunk 1 above)
Comment (jGaboardi) [reply sgeop/artifacts.py:84]: Shorter and used enough to make lines too long. TBH I like `polygons` better, but I was trying to limit lines of code. If you hate it we can change it back.

Comment (martinfleis) [reply sgeop/artifacts.py:84]: I don't mind. I just wouldn't bother.

**Review by martinfleis — APPROVED** [review]

**Pull request merged**